clients/cobol-lsp-vscode-extension/src/services/Middleware.ts

```diff
@@ -25,21 +25,28 @@ export class Middleware {
         return [params.substring(secondDot + 1, lastDot), params.substring(lastDot + 1)];
     }
     constructor(
-        private copybooksPathGenerator: CopybooksPathGenerator,
-        private copybookResolverURI: CopybookURI) {
+        private copybookResolverURI: CopybookURI,
+        private copybookDownloader: CopybookDownloadService) {
     }
 
     public async handleConfigurationRequest(
         params: ConfigurationParams,
         token: CancellationToken,
         next: ConfigurationRequest.HandlerSignature) {
 
-        if (params.items.length === 1) {
+        if (params.items.length > 0) {
             const sectionName = params.items[0].section;
-            if (sectionName.startsWith("broadcom-cobol-lsp.copybook")) {
+            if (sectionName.startsWith("broadcom-cobol-lsp.copybook-resolve")) {
                 const [cobolFileName, copybookName] = Middleware.extractFileAndCopybookNames(sectionName);
                 return [await this.copybookResolverURI.resolveCopybookURI(copybookName, cobolFileName)];
             }
+            if (sectionName.startsWith("broadcom-cobol-lsp.copybook-download")) {
+                for (const item of params.items) {
+                    const [cobolFileName, copybookName] = Middleware.extractFileAndCopybookNames(item.section);
+                    this.copybookDownloader.downloadCopybook(cobolFileName, copybookName);
+                }
+                return [];
```

Comment:
Regarding to this, also better to define return type in method declaration explicitly